src/main/java/org/springframework/data/couchbase/core/query/Meta.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.couchbase.core.query;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+import org.springframework.util.ObjectUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * Meta-data for {@link Query} instances.
+ *
+ * @author Michael Reiche
+
+ */
+public class Meta {
+
+  private enum MetaKey {
+    MAX_TIME_MS("$maxTimeMS"), MAX_SCAN("$maxScan"), COMMENT("$comment"), SNAPSHOT("$snapshot");
```

Comment:
this stuff is mongo related, right? why do we keep this?

Reply:
I missed this when I cleaned out the Meta annotation. I've removed all the mongo-specific elements, I've left just an EXAMPLE in the enum because it can't be empty.  I've also remove the mongo maxTimeMs from the Meta annotation.